server/src/app.ts

```diff
@@ -76,14 +86,13 @@ export const createApp = (): express.Application => {
       res.send('Error occured. Plesae try again.');
     }
   });
-  app.get('/verify_email/:email/:hashed', async (req: ReqI18n, res) => {
-    const email = qs.unescape(req.params.email);
-    const hashed = qs.unescape(req.params.hashed);
+  app.get('/verify_email/:token', async (req: ReqI18n, res) => {
+    const token = qs.unescape(req.params.token);
 
     try {
-      const validated = await validateCredential(email, hashed);
-      if (validated) {
-        await verifyEmail(email);
+      const validated = verify(token, req.appSecret) as VerificationToken;
```

Comment:
Looks like you are using `verify(token, req.appSecret) as VerificationToken` several times.
How about putting this into one function like `validateEmailToken`?

Reply:
Resolved in 119bc1b